vault/environmentvariablescredentialsprovider.go

```diff
@@ -0,0 +1,59 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"log"
+	"os"
+)
+
+// EnvironmentVariablesCredentialsProvider retrieves AWS credentials available in the OS environment variables
+type EnvironmentVariablesCredentialsProvider struct {
```

Comment:
There might be some overlap with the AWS SDK here? Should we consider using [`config.NewEnvConfig`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#NewEnvConfig) and/or [`credentials.StaticCredentialsProvider`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials#StaticCredentialsProvider)

Reply:
Yes, it's a good point. Using `config.NewEnvConfig` allows to be aligned to the logic of the SDK (and hence CLI), and simplifies the code by a great deal. Implemented in [d07990f](https://github.com/99designs/aws-vault/pull/864/commits/d07990fc4d1bba9094483b5ed50af9da0eb3179a) and rebased on latest `master`